diff --git a/rl_coach/env.sh b/rl_coach/env.sh
index b77cf42..4c204c7 100644
--- a/rl_coach/env.sh
+++ b/rl_coach/env.sh
@@ -1,7 +1,7 @@
 export MINIO_ACCESS_KEY=minio
 export MINIO_SECRET_KEY=miniokey
-export AWS_ACCESS_KEY_ID=minio
-export AWS_SECRET_ACCESS_KEY=miniokey
+#export AWS_ACCESS_KEY_ID=minio
+#export AWS_SECRET_ACCESS_KEY=miniokey
 export WORLD_NAME=New_York_Track
 export ROS_AWS_REGION=us-east-1
 export AWS_REGION=us-east-1
